src/elmo_ethercat_sdk/Configuration.cpp

```diff
@@ -19,11 +19,46 @@
  */
 
 #include <iomanip>
+#include <string>
 
 #include "elmo_ethercat_sdk/Configuration.hpp"
 
 namespace elmo {
 
+bool Configuration::sanityCheck(bool silent) {
+#define CHECK_AND_INFORM(bool_condition)                                       \
```

Comment:
I think you could use a function (or lambda function for this)

Reply:
Yes. The string that is printed would however have to be specified manually for each condition in that case.
I think the macro here is OK as it is available only locally and '#undef'-ed at the and of the function.